src/components/require-auth/RequireAuth.jsx

```diff
@@ -0,0 +1,14 @@
+const { Navigate, useLocation } = require("react-router-dom");
+
+const RequireAuth = ({ children }) => {
+  let location = useLocation();
+  const encodedToken = localStorage.getItem("token");
```

Comment:
suggestion: You can create context for auth and store var isUserLogged instead of reading from local storage everytime.